sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobCheckpointData.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Azure.Storage.Blobs.Models;
+
+namespace Azure.Storage.DataMovement.Blobs
+{
+    /// <summary>
+    /// Base class for Blob source and destination checkpoint data
+    /// which contains shared fields.
+    /// </summary>
+    internal abstract class BlobCheckpointData : StorageResourceCheckpointData
+    {
+        /// <summary>
+        /// Schema version.
+        /// </summary>
+        public int Version;
+
+        /// <summary>
+        /// The type of blob.
+        /// </summary>
+        public BlobType BlobType;
+
+        public BlobCheckpointData(int version, BlobType blobType)
+        {
+            Version = version;
+            BlobType = blobType;
+        }
+
+        protected static void CheckSchemaVersion(int expected, int actual)
+        {
+            if (expected != actual)
+            {
+                throw Errors.UnsupportedJobSchemaVersionHeader(actual.ToString());
+            }
+        }
```

Comment:
I mean it was just a check that needed to be done in both implementations of this abstract class and the thought is it might get more complex in the future when we add more versions. I can just duplicate the check for now and remove this.